packages/sirius-web/frontend/sirius-web-application/src/views/library-browser/LibraryBrowser.tsx

```diff
@@ -0,0 +1,143 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { useComponent } from '@eclipse-sirius/sirius-components-core';
+import Container from '@mui/material/Container';
+import Grid from '@mui/material/Grid';
+import Typography from '@mui/material/Typography';
+import { MaterialReactTable, MRT_ColumnDef, MRT_PaginationState, useMaterialReactTable } from 'material-react-table';
+import { memo, useMemo, useState } from 'react';
+import { makeStyles } from 'tss-react/mui';
+import { footerExtensionPoint } from '../../footer/FooterExtensionPoints';
+import { NavigationBar } from '../../navigationBar/NavigationBar';
+import { LibrariesTableProps } from './LibraryBrowser.types';
+import { useLibraries } from './useLibraries';
+import { GQLLibrary } from './useLibraries.types';
+
+const useLibrariesViewStyle = makeStyles()((theme) => ({
+  librariesView: {
+    display: 'grid',
+    gridTemplateColumns: '1fr',
+    gridTemplateRows: 'min-content 1fr min-content',
+    minHeight: '100vh',
+  },
+  main: {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: theme.spacing(5),
+    paddingTop: theme.spacing(3),
+    paddingBottom: theme.spacing(3),
+  },
+  header: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+}));
+
+export const LibraryBrowser = () => {
+  const { classes } = useLibrariesViewStyle();
+  const { Component: Footer } = useComponent(footerExtensionPoint);
+
+  return (
+    <div className={classes.librariesView}>
+      <NavigationBar />
+      <Container maxWidth="xl">
+        <Grid container justifyContent="center">
+          <Grid item xs={8}>
+            <main className={classes.main}>
+              <div className={classes.header}>
+                <Typography variant="h4">Libraries</Typography>
+              </div>
+              <LibrariesTable />
+            </main>
+          </Grid>
+        </Grid>
+      </Container>
+      <Footer />
+    </div>
+  );
+};
+
+const LibrariesTable = memo(({}: LibrariesTableProps) => {
+  const [pagination, setPagination] = useState<MRT_PaginationState>({
+    pageIndex: 0,
+    pageSize: 20,
+  });
+  const { data: libraries, totalLibrariesCount } = useLibraries(pagination.pageIndex, pagination.pageSize);
+
+  const rows: GQLLibrary[] = libraries?.viewer.libraries.edges.map((edge) => edge.node) || [];
```

Comment:
I don't think this works the way you expect.
When the `pageIndex` is updated, the `totalLibrariesCount` will temporarily be `0` (when the request is loading). This value is passed to the table, which updates its pagination state and resets the `pageIndex` value to `0`. 
As a result, you can't ever navigate to the second page of the table.

The previous version of the code was a (bad) workaround to this issue by passing the `rows` and `rowsCount` as props of the component. 

If you want to keep the `useLibraries` hook in this component, I think you should store the `totalLibrariesCount` in a state, and only update it if the query is not loading.